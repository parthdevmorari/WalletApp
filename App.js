import React from 'react';
import {Provider} from "react-redux";
import {configureStore} from './src/Store';
import { persistStore } from "redux-persist";
import { PersistGate } from "redux-persist/es/integration/react";
import Containers from './src/Containers';

const store = configureStore();

class App extends React.Component {
    constructor(props) {
		super(props);
    }
    
    render() {
		const persistor = persistStore(store);
		return (
            <Provider store={store}>
				<PersistGate persistor={persistor} >
						<Containers />
				</PersistGate>
			</Provider>
        );
    }
};

export default App;
