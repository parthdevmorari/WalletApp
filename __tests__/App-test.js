/**
 * @format
 */

 import 'react-native';
 import React from 'react';
 import Enzyme, {shallow, mount} from 'enzyme';
 import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
 import toJson from 'enzyme-to-json';
 import {Provider} from 'react-redux';
 import configureMockStore from 'redux-mock-store';
 import App from '../App';
 
 Enzyme.configure({adapter: new Adapter()});
 
 const mockStore = configureMockStore();
 const store = mockStore({});
 
 describe('Test Main App File', () => {
   test('Test Main App File', () => {
     const wrapper = shallow(
       <Provider store={store}>
         <App />
       </Provider>,
     );
     expect(wrapper.exists()).toBe(true);
   });
   it('Test Main App File', async () => {
     const wrapper = shallow(
       <Provider store={store}>
         <App />
       </Provider>,
     );
     expect(toJson(wrapper)).toMatchSnapshot();
   });
 });
 