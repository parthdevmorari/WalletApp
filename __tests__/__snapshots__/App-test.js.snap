// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App file App  toMatchSnapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <PersistGate
    loading={null}
    persistor={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "flush": [Function],
        "getState": [Function],
        "pause": [Function],
        "persist": [Function],
        "purge": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Containers />
  </PersistGate>
</Provider>
`;

exports[`Test Main App File Test Main App File 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <App />
</ContextProvider>
`;
